themes.go

```diff
@@ -0,0 +1,98 @@
+//go:generate sh sync-themes.sh
+package main
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+var (
+	//go:embed themes.json
+	themesBts []byte
+
+	//go:embed themes_custom.json
+	customThemesBts []byte
+
+	themesOnce sync.Once
+	allThemes  = map[string]Theme{}
+)
+
+func sortedThemeNames() []string {
+	keys := make([]string, 0, len(Themes()))
+	for k := range Themes() {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return strings.ToLower(keys[i]) < strings.ToLower(keys[j])
+	})
+	return keys
+}
+
+// Themes return the list of themes.
+func Themes() map[string]Theme {
+	themesOnce.Do(func() {
+		var themes []windowsTerminalTheme
+		for _, bts := range [][]byte{themesBts, customThemesBts} {
+			if err := json.Unmarshal(bts, &themes); err != nil {
+				fmt.Fprintf(os.Stderr, "could not load themes.json: %s\n", err)
+				os.Exit(1)
+			}
+			for _, theme := range themes {
+				allThemes[theme.Name] = Theme{
+					Background:    theme.Background,
+					Foreground:    theme.Foreground,
+					Selection:     theme.SelectionBackground,
+					Cursor:        theme.CursorColor,
+					CursorAccent:  theme.CursorAccent,
+					Black:         theme.Black,
+					BrightBlack:   theme.BrightBlack,
+					Red:           theme.Red,
+					BrightRed:     theme.BrightRed,
+					Green:         theme.Green,
+					BrightGreen:   theme.BrightGreen,
+					Yellow:        theme.Yellow,
+					BrightYellow:  theme.BrightYellow,
+					Blue:          theme.Blue,
+					BrightBlue:    theme.BrightBlue,
+					Magenta:       theme.Purple,
+					BrightMagenta: theme.BrightPurple,
```

Comment:
If you want, we can change `magenta` and `brightMagenta` to be `purple` / `brightPurple`, so that we can be consistent. It wouldn't be a breaking change but it would make purples of the themes be off.

Reply:
Alternatively we could `sed s/purple/magenta` (when syncing) in the `themes.json` file so that we don't have to do this conversion / renaming in code.

Reply:
I did that initially, but seems like xterm uses magenta instead of purple... there are other differences too, like cursor color and selection background.

Reply:
done